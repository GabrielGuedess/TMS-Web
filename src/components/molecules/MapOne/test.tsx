import { type ReactNode } from 'react';

import { render, waitFor } from '@testing-library/react';

import { MapOne } from '.';

jest.mock('@react-jvectormap/core', () => ({
  VectorMap: ({ children }: { children: ReactNode }) => (
    <div data-testid="Mock VectorMap">{children}</div>
  ),
}));

jest.mock('@react-jvectormap/brazil', () => ({
  brMill: {
    name: 'br_mill',
    content: {
      width: 220_000,
      height: 194_010,
      paths: {
        ac: {
          path: '',
          name: 'Acre',
        },
        pa: {
          path: '',
          name: 'Pará',
        },
        ce: {
          path: '',
          name: 'Ceará',
        },
        ap: {
          path: '',
          name: 'Amapá',
        },
        pi: {
          path: '',
          name: 'Piauí',
        },
        go: {
          path: '',
          name: 'Goiás',
        },
        ba: {
          path: '',
          name: 'Bahia',
        },
        pr: {
          path: '',
          name: 'Paraná',
        },
        se: {
          path: '',
          name: 'Sergipe',
        },
        al: {
          path: '',
          name: 'Alagoas',
        },
        pb: {
          path: '',
          name: 'Paraíba',
        },
        rr: {
          path: '',
          name: 'Roraima',
        },
        ro: {
          path: '',
          name: 'Rondônia',
        },
        am: {
          path: '',
          name: 'Amazonas',
        },
        ma: {
          path: '',
          name: 'Maranhão',
        },
        to: {
          path: '',
          name: 'Tocantins',
        },
        sp: {
          path: '',
          name: 'São Paulo',
        },
        pe: {
          path: '',
          name: 'Pernambuco',
        },
        mt: {
          path: '',
          name: 'MatoGrosso',
        },
        mg: {
          path: '',
          name: 'Minas Gerais',
        },
        es: {
          path: '',
          name: 'Espírito Santo',
        },
        rj: {
          path: '',
          name: 'Rio de Janeiro',
        },
        sc: {
          path: '',
          name: 'Santa Catarina',
        },
        df: {
          path: '',
          name: 'Distrito Federal',
        },
        rs: {
          path: '',
          name: 'Rio Grande do Sul',
        },
        ms: {
          path: '',
          name: 'MatoGrosso do Sul',
        },
        rn: {
          path: '',
          name: 'Rio Grande do Norte',
        },
      },
    },
  },
}));

describe('<MapOne />', () => {
  it('should render the map one correctly', async () => {
    await waitFor(() => {
      const { container } = render(<MapOne />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
