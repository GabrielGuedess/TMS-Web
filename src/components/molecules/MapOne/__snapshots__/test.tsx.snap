// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MapOne /> should render the map one correctly 1`] = `
<div
  class="col-span-12 flex flex-col gap-8 rounded border border-zumthor-100 bg-white-lilac-50 px-7 py-6 shadow-default transition-all dark:border-shark-950 dark:bg-smoke-950 xl:col-span-7"
>
  <h4
    class="mb-2 text-xl font-semibold text-shark-900 dark:text-white-lilac-50"
  >
    Região dos Usuários
  </h4>
  <div
    class="mapOne h-96"
  />
</div>
`;
